File: ruby/m101/hw1-2.rb
require '../lib/mongo_course'

config = {:server => '127.0.0.1', :db => 'm101', :collection => 'funnynumbers'}
MongoCourse.connect(config)

magic = 0

begin
  MongoCourse.find.each do |item|
    if (item['value'] % 3) == 0
      magic = magic + item['value']
    end
  end
rescue Exception => message
  puts message
end

puts "The answer to Homework One, Problem 2 is #{magic.to_i}"

File: node/m101/hw1-2.js
var mongodb = require('mongodb');
var _ = require('underscore');

var db = mongodb.Db('m101', new mongodb.Server('localhost', 27017), {safe: true});

db.open(function(err, client) {
	if(err) {
		console.log('Error opening connection' + err);
	}
	var collection = new mongodb.Collection(client, 'funnynumbers');
	var magic = 0;
	collection.find().toArray(function(err, docs) {
		if (err) {
			console.log(err);
		}
		_.each(docs, function(iter) {
			if (iter.value % 3 === 0) {
				magic += iter.value;
			}
		});
		console.log('The answer to Homework One, Problem 2 is ' + magic);
		client.close();
	});
});
